src/main/java/org/opensearch/security/filter/SecurityRestFilter.java

```diff
@@ -259,6 +259,17 @@ private boolean checkAndAuthenticateRequest(RestRequest request, RestChannel cha
             }
         }
 
+        if (HTTPHelper.containsOBOToken(request)) {
```

Comment:
This doesn't feel like the correct place to put this logic. Would it make sense to put this in the `OnBehalfOfAuthenticator.extractCredentials0` and fail to authenticate within the authenticator if its using the token for a forbidden endpoint?

Also wdyt about updating the OBOAuthenticator to check for the type claim this PR introduces as well?

Reply:
> Would it make sense to put this in the OnBehalfOfAuthenticator.extractCredentials0 and fail to authenticate within the authenticator if its using the token for a forbidden endpoint?

I think I get the idea.I see I think I got the idea. Is that fine to leave `containsOBOToken()` check in the `HTTPHelper`?

> Also wdyt about updating the OBOAuthenticator to check for the type claim this PR introduces as well?

Sure thing! I just added the check and test cases.

Reply:
If a request can be authenticated by the OnBehalfOfAuthenticator than there's no need for the `containsOBOToken()` check because the authenticator extracts the Authorization header and verifies the token in the HTTPHeader.

The reason its best to put this check after authentication is that we can log the attempts to change password or request a new token as suspicious behavior.